macro-rules/src/util.rs

```diff
@@ -0,0 +1,25 @@
+#[macro_export]
+macro_rules! macro_if {
+    (if [] { $($then:tt)* }) => {};
+
+    (if false { $($then:tt)* }) => {};
+
+    (if [$($input:tt)+] { $($then:tt)* }) => {
+        $($then)*
+    };
```

Comment:
I'm kinda confused, but this looks like it ignores the $input arguments? AFAICT it expands identically to `if true`

Reply:
ah—I think actually all of these expect the last two are dead code